packages/kbn-language-documentation-popover/src/components/shared/documentation_navigation.tsx

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+import React from 'react';
+import { css } from '@emotion/react';
+import {
+  EuiFlexItem,
+  EuiFlexGroup,
+  EuiLink,
+  useEuiTheme,
+  EuiFieldSearch,
+  EuiComboBox,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+
+interface DocumentationNavProps {
+  searchText: string;
+  setSearchText: (text: string) => void;
+  onNavigationChange: (selectedOptions: Array<{ label: string }>) => void;
+  filteredGroups?: Array<{ label: string }>;
+  linkToDocumentation?: string;
+  selectedSection?: string;
+}
+
+function DocumentationNav({
+  searchText,
+  setSearchText,
+  onNavigationChange,
+  filteredGroups,
+  linkToDocumentation,
+  selectedSection,
+}: DocumentationNavProps) {
+  const { euiTheme } = useEuiTheme();
+
+  return (
+    <>
+      {linkToDocumentation && (
+        <EuiFlexItem
+          grow={false}
+          css={css`
+            align-items: flex-end;
+            padding-top: ${euiTheme.size.xs};
+          `}
+        >
+          <EuiLink
+            external
+            href={linkToDocumentation}
+            target="_blank"
+            data-test-subj="language-documentation-navigation-link"
+          >
+            {i18n.translate('languageDocumentationPopover.esqlDocsLabel', {
+              defaultMessage: 'View full ES|QL documentation',
+            })}
+          </EuiLink>
+        </EuiFlexItem>
+      )}
+      <EuiFlexGroup gutterSize="none" responsive={false} direction="column">
+        <EuiFlexItem
+          grow={false}
+          css={css`
+            padding: ${euiTheme.size.s} 0;
+          `}
+        >
+          <EuiFieldSearch
+            value={searchText}
+            onChange={(e) => {
+              setSearchText(e.target.value);
+            }}
+            data-test-subj="language-documentation-navigation-search"
+            placeholder={i18n.translate('languageDocumentationPopover.searchPlaceholder', {
+              defaultMessage: 'Search',
```

Comment:
I've tried to Search a bit but I can't understand its mechanism. Seems that search but there is no text highlight and the help is scrolled to the top of the groups and not to the first found match.
Since this looks more like a command guideline, do we really need the search? can't we just search by the command name by using the other dropdown?

Reply:
It works as currently so I didnt change the mechanism. You used to allow search only in the title (command, functions etc) but it was very limited so now it also searches in the description too. So if I type DROP it will show the DROP command but also sections with DROP examples. We think that it makes it more powerful. We are also thinking of highlighting but is not at the scope of this PR.

Reply:
so if you just allow to search for the title I believe this is another point in favour of get rid of the search input in favour of the command dropdown that already provides the same capability

Reply:
We are allowing search for title and description. It used to be only title some minors ago.